src/python/pants/backend/javascript/subsystems/nodejs.py

```diff
@@ -268,6 +308,35 @@ def immutable_digest(self) -> dict[str, Digest]:
         )
 
 
+async def add_corepack_shims_to_digest(
+    binaries: NodeJSBinaries, tool_shims: BinaryShims, corepack_env_vars: EnvironmentVars
+) -> Digest:
+    directory_digest = await Get(Digest, CreateDigest([Directory("._corepack")]))
+    binary_digest = binaries.digest if binaries.digest else EMPTY_DIGEST
+    input_digest = await Get(Digest, MergeDigests((directory_digest, binary_digest)))
```

Comment:
There is an (unlikely) situation here that could break things. If a user uses a bootstrapped nodejs that _doesnt ship with corepack_ (no LTS versions of this dist exists https://nodejs.dev/en/about/releases/, corepack is part of v14 dist), the "corepack enable" invocation on line 324 will surely not behave nicely.

I'm reluctant to implement support for versions <14 as that is past maintenance, but maybe better error messaging is warranted than a filenotfound error from the engine?

Reply:
Yea, that seems reasonable.